### DISCUSSION TITLE
Is it possible to clone or save/restore engine state?

### DISCUSSION THREAD

**Discussion by KarolKolenda** [sebastienros/jint#1060] [Q&A]
I've got relatively long initial code and then I use the engine to execute a single statement which can affect engine state. I would like to speed things up by not re-executing initial code all the time. Would it be possible to:
a) quickly clone the instance of engine with it entire state or
b) save and restore quickly engine state?

Many thanks,
Karol

**Comment by pillepalle1** [comment]
+1
It would be great if Engine would implement ICloneable

**Comment by lahma** [comment]
I'm wokring on a PR https://github.com/sebastienros/jint/issues/788 that should serve this purpose by allowing a temporary context to run commands in that can then be cleaned.
